tests/select.spec.ts

```diff
@@ -1,20 +1,204 @@
+  // show the selcted value in the input
+  expect(await input.inputValue()).toEqual('one')
+});
+
+test('Given a select with a placeholder, that should be visible in the input', async ({page}) => {
+  const select = page.getByTestId('basic-select')
+  await expect(select).toBeVisible()
+
+  await expect(select.locator('input').first()).toHaveAttribute('placeholder', 'All')
+})
+
+test('Setting withbutton to true, with a button text and icon, the text and icon should be rendered as a final element', async ({page}) => {
+  const select = page.getByTestId('select-with-button')
+  await expect(select).toBeVisible()
+
+  await select.locator('input').first().focus()
+
+  const button = select.locator('v-select-button') 
+  await expect(button).toBeVisible()
+  await expect(button).toHaveText('Add new option')
+  await expect(button.locator('v-icon > i')).toHaveClass(/icon-add/)
+
+})
+
+test('If a select component has disabled=True the element should be disabled', async ({page}) => {
+  const select = page.getByTestId('select-disabled')
+  const input = select.locator('input').first()
+  await input.focus()
+  await input.type('new option')
+  // options do not load and the actual one isn't visible
+  expect (select.locator('.options-container').first()).not.toBeVisible()
+  expect(await input.inputValue()).toEqual('')
+})
+
+test('If the select component has readonly value, then it should be readonly', async ({page}) => {
+  const select = page.getByTestId('select-readonly')
+  const input = select.locator('input').first()
+  await input.focus()
+  await input.type('new option')
+  // options do not load and the actual one isn't visible
+  expect (select.locator('.options-container').first()).not.toBeVisible()
+  expect(await input.inputValue()).toEqual('')
+})
+
+test('If the select has attribute exact, the user can only type a value from the dropdown', async ({page}) => {
+  const select = page.getByTestId('select-exact')
+  const input = select.locator('input').first()
+
+  await input.focus()
+  await input.type('not an option')
+
+  await page.keyboard.press('Enter')
+
+  expect(await input.inputValue()).toEqual("")
+
+  // now type a valid option
+  await input.type('one')
+  await page.keyboard.press('Enter')
+  expect(await input.inputValue()).toEqual('one')
+
+})
+
+test('No state but tooltip applied should render info icon with tooltip text', async ({page}) => {
+  const select = page.getByTestId('select-default-tooltip')
+  const tooltip = select.locator('label > div > v-tooltip').first()
+  await expect(tooltip.locator('.icon-info-outline')).toBeVisible()
+
+  // text is not visible
+  await expect(tooltip.getByText('default tip').first()).not.toBeVisible()
+  // make
+  await tooltip.hover()
+  await expect(tooltip.getByText('default tip').first()).toBeVisible()
+
+})
+
+test('Given info state and tooltip value, there should be an info tooltip next to the select label', async ({page}) => {
```

Comment:
not sure if this is necessary, but should we test the info state here? maybe checking for the right class or color or something.